src/control/cmd/daos_agent/mgmt_rpc.go

```diff
@@ -170,10 +175,24 @@ func (mod *mgmtModule) handleGetAttachInfo(reqb []byte, pid int32) ([]byte, erro
 		return nil, err
 	}
 
-	return mod.aiCache.getResponse(numaNode)
+	cacheResp, err := mod.aiCache.getResponse(numaNode)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
In this case it has to come after. The first two cases we know the information is there since its already cached. This instance of it can fail and we don't want to register the process if we fail to get the attachment information.

Reply:
Ah, got it.